component/prometheus/scrape/scrape.go

```diff
@@ -295,3 +299,20 @@ func convertLabelSet(tg discovery.Target) model.LabelSet {
 	}
 	return lset
 }
+
+func getTargetsGaugeMetric(registerer client_prometheus.Registerer) *client_prometheus.GaugeVec {
+	targetsGauge := client_prometheus.NewGaugeVec(client_prometheus.GaugeOpts{
+		Name: "agent_prometheus_scrape_targets_gauge",
+		Help: "A gauge of all scrape targets this component is configured to scrape",
+	}, []string{})
+	err := registerer.Register(targetsGauge)
+	if err != nil {
+		if existing, ok := err.(client_prometheus.AlreadyRegisteredError); ok {
+			targetsGauge = existing.ExistingCollector.(*client_prometheus.GaugeVec)
+		} else {
+			// Same behavior as MustRegister if the error is not for AlreadyRegistered
+			panic(err)
```

Comment:
Did you come across this error? That seems like a smell, all of these metrics should be unique.